hatch_build.py

```diff
@@ -0,0 +1,83 @@
+from hatchling.builders.hooks.plugin.interface import BuildHookInterface
+from hatchling.metadata.plugin.interface import MetadataHookInterface
+
+import tomlkit
+import pathlib
+
+
+class PinVersionMetadataHook(MetadataHookInterface):
+    """Update the dependency versions so they are pinned metadata of the wheel artefact produced by a wheel."""
```

Comment:
Can you add a comment summarising when this gets called, for people who aren't intimately familiar with the callback model for hatch?